Use new APIs for frisbee-1.4 and pgprs-2.0; add get...devices 'ALL' option

rserwin1 opened the pull request:
These changes support new APIs of the frisbee-1.4 and pgprs-2.0 packages.  They are invoked by the commands, frisbee and pgprs.  They take a single argument for a specific action.  The connectwizard scripts will use them if present.  If an older version of frisbee is installed, the old interface is used.  For pgprs, only the new package is supported, because the old version should be removed from woof-CE and a new pgprs package option added to woof-CE.

Although the results from get_modem_alternate_device omit an argument-specified device name, the new option requests that all found devices be returned, as needed by the new frisbee and pgprs.

01micko commented:
Hello @rserwin1 , welcome aboard!

I'll merge this but what about adding the frisbee package to `rootfs-packages`?

Just take a look at any of the packages - it is just the standard .pet structure.

Cheers!

rserwin1 commented:
Mick,
Thanks for the suggestion.  I would like to add frisbee and pgprs, if 
that is what is appropriate.  However, I cannot figure out how to do 
it.  I expanded the frisbee pet package on my PC and attempted to upload 
the frisbee-1.4 directory.  I saw a little message that the file was 
empty.  Then I tried creating a file. frisbee-1.4, but it wanted me to 
enter text.  So, how do I upload the entire directory to my fork of woof_CE?

Re the merge: I seem to have entered duplicate requests.  I see your 
name on the one with the correct subject.  But everywhere the files are 
shown, it is the duplicate, "Add files via upload"!  Can you fix that so 
that the correct request is official..  And delete the bad duplicate?  
How should I have deleted the accidental second request?
Richard

Mick Amadio wrote:
>
> Hello @rserwin1 <https://github.com/rserwin1> , welcome aboard!
>
> I'll merge this but what about adding the frisbee package to 
> |rootfs-packages|?
>
> Just take a look at any of the packages - it is just the standard .pet 
> structure.
>
> Cheers!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/puppylinux-woof-CE/woof-CE/pull/778#issuecomment-221152792>
>

01micko commented:
Hi Richard,

@zigbert is the main one here that uploads stuff via the GitHub web interface so perhaps he can offer advice. I always use git via CLI. I haven't a clue about the web interface! :fearful: 

Here is a brief CLI tutorial https://help.github.com/articles/set-up-git/

- Fork the woof-CE repo then clone _your_ fork locally.

- make the changes, add, delete files/directories as needed

- run `git add .` This stages your changes for commit

- run `git commit` and enter the commit message in `mp` or whatever text editor you have. 

- push to _your_ fork `git push origin my-fork`

HTH

> Re the merge: I seem to have entered duplicate requests.  I see your
name on the one with the correct subject.  But everywhere the files are
shown, it is the duplicate, "Add files via upload"!  Can you fix that so
that the correct request is official..  And delete the bad duplicate?
How should I have deleted the accidental second request?
Richard

Seems ok here. Maybe it's just on you fork? If so just delete that branch and create a new one, or if unsure just delete your fork and create a new one.

Cheers!

zigbert commented:
AFAIK, the web interface cannot upload a directory - only files - one by one.
I use the cli-way when uploading packages.

zigbert commented:
I found a way to upload via web interface. Will make a blogpost about it

01micko commented:
By the way.... ( @rserwin1 )

> I would like to add frisbee and pgprs, if that is what is appropriate.

All packages in `rootfs-packages` are optional for the distro builder and are tested against the package list for their existence and/or dependencies (though some manual intervention in 3builddistro[-Z] may be needed on my part to test for wpa-supplicant etc. No trouble at all for me though).

peabee commented:
Hi Richard
Just tried a build with a fresh woof-ce...
There is a frisbee directory in rootfs-packages BUT with no real frisbee content....
The directory shouldn't be there until it contains the full frisbee package....
It is still necessary to include the frisbee-1.4 pet explicitly in the build.
Cheers
Peter

ninaholic commented:
Hi Richard,

I might be wrong, but I think line 174 in connectwizard:

> if [ -x /usr/sbin/pgprs -o -x /usr/sbin/pgprs-shell ];then #160120

Will always equal true, because pgprs-shell is part of rootfs-skeleton? Since your pgprs2 is not included yet, this might cause issues when the user selects the "Wireless GPRS modem" button in connectwizard when it reaches pgprs-setup unless your pgprs2 is added?

I'm curious, in frisbee it says "Jemimah's Wizard". Are you also known as Jemimah, or are there two people creating frisbee? Also: on my dads internet, the SSID shows up as "FRED'S" instead of "FRED'S FIBE" even when I expand the field in frisbee... is this intentional or a bug with spaces on my machine? :D

I think having your tools in Woof-CE is a nice idea. ;)

01micko commented:
Ok, I think I'll merge this in about 14 hours from now. Any objections please post and I'll consider those. Any bugs can be sorted out as we go, such as @ninaholic 's dad's spaces bug. 

@peabee , since @zigbert did his 'showcase' commits that bug you are experiencing is apparent. It will be sorted once this PR is merged or otherwise. Stay tuned.

Cheers!

rserwin1 commented:
@ninaholic 
Thanks for pointing out the problem with line 174.  I have removed the pgprs-shell part and also removed file /usr/sbin/pgprs-shell.  I could not find the all-firmware/pgprs directory to remove it, as well.  These changes are my new branch, pgprs-branch, to try that means of keeping the content of push requests separate.

Re frisbee, Jemimah left the forum some years ago, so I "took possession of it because so many people seemed to like it and want it.  I could not have created it myself.  But as a former maintenance programmer, I felt I could fix bugs and improve some parts -- and then add a knock-off from pgprs, to it.  I want to keep Jemimah's name on it, but also want frisbee to be as good as those interested can make it.

Re "FRED'S FIBE", I think you found a bug!  The space certainly is the first thiing to look into.  On my to-do list.  Thanks for mentioning it.

I expect to make a pgprs pull requist shortly.
R

01micko commented:
See how tired I was yesterday? lol. I meant that 'merge' statement to be for #779 which I just merged.
